sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java

```diff
@@ -247,15 +480,12 @@ public void setEvalGroupsForNode(String nodeId, Set<String> evalGroupIds) {
         Set<String> s = new HashSet<String>();
         if (evalHierarchyProvider != null) {
             s = evalHierarchyProvider.getEvalGroupsForNode(nodeId);
-        } else {
-            EvalGroupNodes egn = getEvalGroupNodeByNodeId(nodeId);
-            if (egn != null) {
-                String[] evalGroups = egn.getEvalGroups();
-                for (int i = 0; i < evalGroups.length; i++) {
-                    s.add(evalGroups[i]);
-                }
-            }
         }
+
+        // Support for resolving eval groups based on hierarchy node rules.
+        // Hierarchy rules should be obeyed regardless of if an external provider is present or not (supplemental)
+        s.addAll( getEvalGroupsForNodeSectionAware( nodeId ) );
```

Comment:
I'm also curious for opinions on the logic here (and in the other spots marked "supplemental"), on whether this behaviour should be applied regardless of having a provider present or not.

My opinion is that the hierarchy rules logic should be applied regardless, but I want to be sure others are of the same opinion.